lib/server/api.rs

```diff
@@ -0,0 +1,305 @@
+//! API implementation to be used by the server
+
+use crate::rlwp::Job;
+use crate::server::user::User;
+use crate::utils::errors::{RLWServerError, GENERAL_SERVER_ERR, NO_UUID_JOB};
+use crate::utils::job_processor_api::{job_processor_service_server::JobProcessorService, *};
+use crate::utils::tls;
+
+use std::{
+    collections::HashMap,
+    sync::{Arc, Mutex},
+};
+use tokio::sync::mpsc as tokio_mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+use tonic::{Request, Response, Status};
+
+// TODO: Update this to be a more appropriate size:
+//
+// Run multiple client streaming tests to determine the maximum number of items
+// that appeared in the buffer at once. Set the STREAM_BUFFER_SIZE to be
+// 1.5 times larger than this.
+const STREAM_BUFFER_SIZE: usize = 4096;
+
+/// Handles the incoming gRPC job process requests and
+/// stores user data related to these requests
+pub struct JobProcessor {
+    // Stores all users and their jobs.
+    // Maps usernames to users structs.
+    user_table: Arc<Mutex<HashMap<String, Arc<User>>>>,
+}
+
+#[tonic::async_trait]
+/// gRPC Service Implementation
+impl JobProcessorService for JobProcessor {
+    /// Start a new process job
+    async fn start(
+        &self,
+        request: Request<StartRequest>,
+    ) -> Result<Response<StartResponse>, Status> {
+        log::info!("Start Request");
+
+        // Authorize user and get user object
+        let user_id = tls::authorize_user(request.peer_certs())
+            .map_err(Status::unknown)?
+            .ok_or_else(|| Status::unknown("User is not authorized"))?;
```

Comment:
```suggestion
            .ok_or_else(|| Status::unauthorized("User is not authorized"))?;
```

Here and in the other functions.

Reply:
Resolved.

Reply:
Resolved. Updated all functions.